tests/unit/crsAdapter/TomaSPCAdapter.test.js

```diff
@@ -22,73 +22,20 @@ describe('TomaSPCAdapter', () => {
-        let expectedDest = 'https://www.em1.sellingplatformconnect.amadeus.com';
-
+    it('connect() should result in error when no connection url is detected', (done) => {
         adapter.connect().then(() => {
-            let scriptElement = documentHeadAppendChildSpy.calls.mostRecent().args[0];
-
-            expect(scriptElement.src).toBe(expectedSrc);
-            expect(TomaSPCConnection.dest).toBe(expectedDest);
-            done();
-        }, (error) => {
-            done.fail(error);
-        });
-    });
-
-    it('connect() with option.crsUrl should result in correct script.src', (done) => {
-        let expectedSrc = 'https://crsurl/ExternalCatalog.js';
-        let expectedDest = 'https://crsUrl';
-
-        adapter.connect({crsUrl: 'http://crsUrl'}).then(() => {
-            let scriptElement = documentHeadAppendChildSpy.calls.mostRecent().args[0];
-
-            expect(scriptElement.src).toBe(expectedSrc);
-            expect(TomaSPCConnection.dest).toBe(expectedDest);
-            done();
-        }, (error) => {
-            done.fail(error);
-        });
-    });
-
-    it('connect() with option.env "test" should result in correct script.src', (done) => {
-        let expectedSrc = 'https://acceptance.emea1.sellingplatformconnect.amadeus.com/ExternalCatalog.js';
-        let expectedDest = 'https://acceptance.emea1.sellingplatformconnect.amadeus.com';
-
-        adapter.connect({env: 'test'}).then(() => {
-            let scriptElement = documentHeadAppendChildSpy.calls.mostRecent().args[0];
-
-            expect(scriptElement.src).toBe(expectedSrc);
-            expect(TomaSPCConnection.dest).toBe(expectedDest);
-            done();
+            done.fail('unexpected result');
         }, (error) => {
-            done.fail(error);
-        });
-    });
-
-    it('connect() with option.env "prod" should result in correct script.src', (done) => {
-        let expectedSrc = 'https://www.em1.sellingplatformconnect.amadeus.com/ExternalCatalog.js';
-        let expectedDest = 'https://www.em1.sellingplatformconnect.amadeus.com';
-
-        adapter.connect({env: 'prod'}).then(() => {
-            let scriptElement = documentHeadAppendChildSpy.calls.mostRecent().args[0];
-
-            expect(scriptElement.src).toBe(expectedSrc);
-            expect(TomaSPCConnection.dest).toBe(expectedDest);
+            expect(error.toString()).toBe('Error: could not detect any Amadeus SeCo URL to connect to');
             done();
-        }, (error) => {
-            done.fail(error);
         });
     });
 
-    it('connect() with URL parameter crs_url should result in correct script.src', (done) => {
-        let expectedSrc = 'https://crsurl/ExternalCatalog.js';
-        let expectedDest = 'https://crsUrl';
-
-        window.history.replaceState({}, '', '?crs_url=http://crsUrl');
+    it('connect() with option.connectionUrl should result in correct script.src', (done) => {
+        let expectedSrc = 'https://conn-url/ExternalCatalog.js';
```

Comment:
Doesn't matter, but I'm a huge fan of [RFC 2606](https://tools.ietf.org/html/rfc2606) URLs.

Reply:
Ah, that is a nice idea.